cmd/frontend/graphqlbackend/send_test_email.go

```diff
@@ -8,11 +8,34 @@ import (
 	"github.com/sourcegraph/log"
 
 	"github.com/sourcegraph/sourcegraph/internal/auth"
+	"github.com/sourcegraph/sourcegraph/internal/conf"
 	"github.com/sourcegraph/sourcegraph/internal/txemail"
 	"github.com/sourcegraph/sourcegraph/internal/txemail/txtypes"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+	"github.com/sourcegraph/sourcegraph/lib/pointers"
+	"github.com/sourcegraph/sourcegraph/schema"
 )
 
-func (r *schemaResolver) SendTestEmail(ctx context.Context, args struct{ To string }) (string, error) {
+type SMTPAuthType string
+
+type smtpConfig struct {
+	Host            string       `json:"host"`
+	Port            int32        `json:"port"`
+	Authentication  SMTPAuthType `json:"authentication"`
+	Username        *string      `json:"username,omitempty"`
+	Password        *string      `json:"password,omitempty"`
+	Domain          *string      `json:"domain,omitempty"`
+	NoVerifyTLS     *bool        `json:"noVerifyTLS,omitempty"`
+	EmailAddress    string       `json:"emailAddress"`
+	EmailSenderName *string      `json:"emailSenderName,omitempty"`
+}
+
+var randomUUID = uuid.NewRandom
+
+func (r *schemaResolver) SendTestEmail(ctx context.Context, args struct {
+	To     string
+	Config *smtpConfig
+}) (string, error) {
```

Comment:
What do you think about creating a concrete struct instead of inline-ing the struct definition?

```suggestion
type SendTestEmailArgs struct {
  To         string
  Config *smtpConfig
}

func (r *schemaResolver) SendTestEmail(ctx context.Context, args SendTestEmailArgs) (string, error) {
```

Reply:
Since it's used only once, I think inlining is fine. This is a [pattern I also noticed elsewhere in the code](https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph@main/-/blob/cmd/frontend/graphqlbackend/access_requests.go?L113).

Reply:
If you want to test this resolver without going through the GQL layer, it would mean repeating that struct all the time, I think the fact that we have it is that we were lazy not that it's a great practice

Not a big deal, but I tend to agree with Bolaji here.